src/app/pages/payment/tokenized-card.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { PaymentService } from 'src/app/services/payment-service';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { Location } from '@angular/common';
+
+@Component({
+  templateUrl: './tokenized-card.component.html',
+  styleUrls: ['./payment.component.scss']
+})
+export class TokenizedCardComponent implements OnInit {
+  cardForm = new FormGroup({
+    amount: new FormControl('', [Validators.required, Validators.min(0)]),
+    purpose: new FormControl('', Validators.required)
+  });
+
+  constructor(
+    private paymentService: PaymentService,
+    private router: Router,
+    private spinner: NgxSpinnerService,
+    private toastr: ToastrService,
+    private location: Location
+    ) { }
+
+  ngOnInit() {	}
+  onSubmit(): void {
+    const payload = {
+      amount: this.cardForm.value.amount,
+      purpose: this.cardForm.value.purpose
+    };
+    this.spinner.show();
+    this.paymentService.payWithTokenizedCard(payload).subscribe(
```

Comment:
Hello @Victoradukwu , did you explicitly handle unsubscribing from the the paymentService ?

Reply:
@malep2007 In a **Pluralsight** video by Deborah Kurata, I learnt it is not necessary to unsubscribe from a **http** service since it completes automatically. I am not able to specifically point to the video timestamp but this **Stackoverflow** post discussed it extensively:  https://stackoverflow.com/questions/35042929/is-it-necessary-to-unsubscribe-from-observables-created-by-http-methods